### ISSUE TITLE
Could you tell me your GPU memory size?

### ISSUE THREAD

**Issue by Ancisky** [Panda0406/Zero-shot-knowledge-graph-relational-learning#4]
Hey, I encountered some problems in the process of reproducing your paper experiments. It‘s warning "Out of Memory“ when I use the NELL dataset to running such loop ”for relname in self.train_tasks.keys():" in file "trainer. py". So I want to understand the GPUs use in your experiments. Please, It's important for me.

**Comment by Panda0406** [comment]
Sorry for the late reply.

For the pretraining stage:
![image](https://user-images.githubusercontent.com/10203118/87218665-51e22d80-c387-11ea-9d86-c1319d2d2406.png)

For the adversarial training stage:
![image](https://user-images.githubusercontent.com/10203118/87218671-5ad2ff00-c387-11ea-8f8a-e7d10cb97bd2.png)

**Comment by Ancisky** [comment]
> Sorry for the late reply.
> 
> For the pretraining stage:
> ![image](https://user-images.githubusercontent.com/10203118/87218665-51e22d80-c387-11ea-9d86-c1319d2d2406.png)
> 
> For the adversarial training stage:
> ![image](https://user-images.githubusercontent.com/10203118/87218671-5ad2ff00-c387-11ea-8f8a-e7d10cb97bd2.png)

Thanks for your reply!! Now there's another problem, do I still run "trainer.py" when I use Wiki dataset? I'm having some trouble about dimension while the code runs to pretrain_featrue_extractor and adversarial training. Or dose it need no pretrain at all when running Wiki dataset?
There is my parameters in “args.py”: 
`embed_dim: 50  w_dim: 50  ep_dim: 200  noise_dim: 15  D_batch_size: 256  G_batch_size: 256`
Do you have some suggestion about args‘s dimension parameters? 
Sorry to trouble you, I am looking forward to your reply.

**Comment by Panda0406** [comment]
Below is my setting for Wiki-data. Hoping that will help you.

###########
grad_clip: 5.0
noise_dim: 15
dataset: Wiki
D_epoch: 5
test_sample: 20
seed: 0
pretrain_margin: 10.0
G_epoch: 1
weight_decay: 0.0
pretrain_few: 30
REG_W: 0.001
train_times: 6000
no_meta: False
pretrain_batch_size: 128
embed_dim: 50
G_SN: False
pretrain_subepoch: 30
D_SN: False
save_path: models/intial
test: False
REG_Wz: 0.0001
gan_batch_rela: 8
embed_model: DistMult
pretrain_loss_every: 200
loss_every: 200
dropout: 0.5
w_dim: 50
aggregate: max
D_batch_size: 64
pretrain_times: 7000
device: 0
prefix: intial
max_neighbor: 50
lr_G: 0.0001
lr_E: 0.0005
lr_D: 0.0001
fine_tune: False
G_batch_size: 64
###########

**Comment by Ancisky** [comment]
> Below is my setting for Wiki-data. Hoping that will help you.
> 
> ###########
> grad_clip: 5.0
> noise_dim: 15
> dataset: Wiki
> D_epoch: 5
> test_sample: 20
> seed: 0
> pretrain_margin: 10.0
> G_epoch: 1
> weight_decay: 0.0
> pretrain_few: 30
> REG_W: 0.001
> train_times: 6000
> no_meta: False
> pretrain_batch_size: 128
> embed_dim: 50
> G_SN: False
> pretrain_subepoch: 30
> D_SN: False
> save_path: models/intial
> test: False
> REG_Wz: 0.0001
> gan_batch_rela: 8
> embed_model: DistMult
> pretrain_loss_every: 200
> loss_every: 200
> dropout: 0.5
> w_dim: 50
> aggregate: max
> D_batch_size: 64
> pretrain_times: 7000
> device: 0
> prefix: intial
> max_neighbor: 50
> lr_G: 0.0001
> lr_E: 0.0005
> lr_D: 0.0001
> fine_tune: False
> G_batch_size: 64
> ###########

Thank you very much for your reply!!! I have been a little busy this week, so I haven't had a chance to express my thanks. I successfully completed the experiment in the paper. I think this job has excellent research prospects!!!

**Comment by ren-1247** [comment]
> > Below is my setting for Wiki-data. Hoping that will help you.
> > ###########
> > grad_clip: 5.0
> > noise_dim: 15
> > dataset: Wiki
> > D_epoch: 5
> > test_sample: 20
> > seed: 0
> > pretrain_margin: 10.0
> > G_epoch: 1
> > weight_decay: 0.0
> > pretrain_few: 30
> > REG_W: 0.001
> > train_times: 6000
> > no_meta: False
> > pretrain_batch_size: 128
> > embed_dim: 50
> > G_SN: False
> > pretrain_subepoch: 30
> > D_SN: False
> > save_path: models/intial
> > test: False
> > REG_Wz: 0.0001
> > gan_batch_rela: 8
> > embed_model: DistMult
> > pretrain_loss_every: 200
> > loss_every: 200
> > dropout: 0.5
> > w_dim: 50
> > aggregate: max
> > D_batch_size: 64
> > pretrain_times: 7000
> > device: 0
> > prefix: intial
> > max_neighbor: 50
> > lr_G: 0.0001
> > lr_E: 0.0005
> > lr_D: 0.0001
> > fine_tune: False
> > G_batch_size: 64
> > ###########
> 
> Thank you very much for your reply!!! I have been a little busy this week, so I haven't had a chance to express my thanks. I successfully completed the experiment in the paper. I think this job has excellent research prospects!!!

Hello, do you know where the relationship is expressed Network.py File neighbor_encoder function, I still can't understand it

def neighbor_encoder(self, connections, num_neighbors):
        '''
        connections: (batch, 200, 2)
        num_neighbors: (batch,)
        '''
        num_neighbors = num_neighbors.unsqueeze(1)
        entities = connections[:,:,1].squeeze(-1)
        ent_embeds = self.dropout(self.symbol_emb(entities)) # (batch, 50, embed_dim)
        concat_embeds = ent_embeds

        out = self.gcn_w(concat_embeds)
        out = torch.sum(out, dim=1) # (batch, embed_dim)
        out = out / num_neighbors
        return out.tanh()

**Comment by Filco306** [comment]
@ren-1247 @Ancisky Did you get it to work with those parameter settings? I get a dimensionality issue in the code, so I cannot run through a complete run on the Wiki dataset. Are you using 200 dimensions in the Generator and the Discriminator on the Wiki dataset? For me, that does not work. However, changing the dimension to 100 by using `ep_dim` as an additional argument to both of these makes it work. I am running it right now, so I do not know if it will reproduce the results of the paper, but I look forward to hearing your replies. Perhaps @Panda0406 has something to say here; did you use 200 or 100 dimensions in the Discriminator and Generator for the Wiki-ZS dataset? 

Cheers
Filip
